### DISCUSSION TITLE
InvalidAuthenticityToken Can't verify CSRF  with rails 7.1.2.

### DISCUSSION THREAD

**Discussion by rajanverma-me** [activeadmin/activeadmin#8189] [Help]
I have a very basic Rails 7.1.2 fresh installed app. I added active admin but as usual, it all worked fine in development. 
The problem happened when I pushed the app to production. 

### Expected behavior

The normal login should work. 

### Actual behavior

The admin login gives me the below error. I tried adding `skip_before_action :verify_authenticity_token` but then it won't give any error and simply redirects to the login page again. I have checked several times the password is correct. 

```
Processing by ActiveAdmin::Devise::SessionsController#create as HTML
App 1353734 output: I, [2023-12-28T11:24:56.673866 #1353734]  INFO -- : [9800ff02-cc91-452d-9820-d150f98918ac]   Parameters: {"authenticity_token"=>"[FILTERED]", "admin_user"=>{"email"=>"rajan@aarvy.me", "password"=>"[FILTERED]", "remember_me"=>"0"}, "commit"=>"Login"}
App 1353734 output: W, [2023-12-28T11:24:56.674428 #1353734]  WARN -- : [9800ff02-cc91-452d-9820-d150f98918ac] Can't verify CSRF token authenticity.
App 1353734 output: I, [2023-12-28T11:24:56.675201 #1353734]  INFO -- : [9800ff02-cc91-452d-9820-d150f98918ac] Completed 422 Unprocessable Entity in 1ms (ActiveRecord: 0.0ms | Allocations: 334)
App 1353734 output: E, [2023-12-28T11:24:56.676495 #1353734] ERROR -- : [9800ff02-cc91-452d-9820-d150f98918ac]   
App 1353734 output: [9800ff02-cc91-452d-9820-d150f98918ac] ActionController::InvalidAuthenticityToken (Can't verify CSRF token authenticity.):
App 1353734 output: [9800ff02-cc91-452d-9820-d150f98918ac]   
App 1353734 output: [9800ff02-cc91-452d-9820-d150f98918ac] actionpack (7.1.2) lib/action_controller/metal/request_forgery_protection.rb:293:in `handle_unverified_request'
App 1353734 output: [9800ff02-cc91-452d-9820-d150f98918ac] actionpack (7.1.2) lib/action_controller/metal/request_forgery_protection.rb:388:in `handle_unverified_request'
```

![image](https://github.com/activeadmin/activeadmin/assets/10785334/33baa40f-94dd-4d5b-94a9-0c5f8786fb0d)

I tried downgrading 3.1 but the error is still the same. I can't downgrade it further because of rail dependency.
I've checked the authenticity token is getting passed in login requests so generation is also not the problem. 

### How to reproduce

Just a very straightforward setup with no additions. The only thing I have done differently is I have initially installed active admin with `--skip users` option and then again installed it with devise users. But I don't think that would be the issue.

**Comment by abidiqbalk** [comment]
Facing same issue with rails 6.1 and active admin 2.14.0. Login on development environment is working fine but on heroku its giving "Can't verify CSRF token authenticity". Any help would highly be appreciated.

**Comment by tagliala** [comment]
Ref: https://github.com/activeadmin/activeadmin/issues/3660#issuecomment-2108608717
